### ISSUE TITLE
Invalid typescript syntax generated

### ISSUE THREAD

**Issue by aabedraba** [prisma/prisma#5740]
## Bug description

When generating prisma client types, a wrong syntax is generated in the module type definition. 

The error steps from trying to use template string to wrap a type. 

This is different from issue #3194 which refers to wrong syntax generation when index column name contains invalid javascript characters.

## How to reproduce

I created a repo here https://github.com/aabedraba/prisma-typescript-error that you can download and reproduce the error with. 

Just need to 
1. Install modules `npm i`
2. Run prisma generate `npx prisma generate`
3. Compile the code with typescript `npx tsc``

## Expected behavior

Correctly generated client types to be able to use the library. 

## Prisma information

``` prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model User {
  id   Int    @id @default(autoincrement())
  name String
}
```

## Environment & setup

- OS: Mac OS - M1 
- Database: PostgreSQL
- Node.js version: v.14.15.4
- Prisma version:
```
prisma               : 2.17.0
@prisma/client       : 2.17.0
Current platform     : darwin
Query Engine         : query-engine 3c463ebd78b1d21d8fdacdd27899e280cf686223 (at node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli 3c463ebd78b1d21d8fdacdd27899e280cf686223 (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core 3c463ebd78b1d21d8fdacdd27899e280cf686223 (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt 3c463ebd78b1d21d8fdacdd27899e280cf686223 (at node_modules/@prisma/engines/prisma-fmt-darwin)
Studio               : 0.353.0
```

**Comment by Jolg42** [comment]
So I think the error comes from the TypeScript version you're using 3.x

Upgrading to 4.1 should fix it.

We're planning to have a message to help about that https://github.com/prisma/prisma/issues/5728

**Comment by timsuchanek** [comment]
Faster than light @Jolg42.
Yes, I think that is it! You need to upgrade to TypeScript `4.1` or later.

**Comment by aabedraba** [comment]
> Upgrading to 4.1 should fix it.

Yes this fixes it. @timsuchanek created a PR in the repo and it worked. 

I just saw that release https://github.com/prisma/prisma/releases/tag/2.14.0 indicates that prisma needs TS 4.1^ as breaking change. 

It's going to be hard to upgrade typescript version in a big TS codebase. Just doing it show right now around 300 errors (and that's from 3.9 to 4.1). We'll keep using prisma 2.10 for now and also keep an eye at https://github.com/prisma/prisma-client-js/issues/949#issuecomment-745166334 as we encountered this issue after running `prisma introspect` when migrating to the new version. 

Thank you!
